src/components/Steps.tsx

```diff
@@ -24,11 +25,24 @@ const Steps = () => {
 
     const handleConnectWallet = () => popupsActions.setIsConnectorsWalletOpen(true)
 
+    const sendConnectWalletEvent = async (walletAddress: string): Promise<void> => {
+        const time = new Date().toDateString()
+        const message = `Sign to verify your address and access our points rewards program ${time}`
+        const signature = await provider?.getSigner().signMessage(message)
+        await Fuul.sendConnectWallet({
+            address: walletAddress,
+            signature,
+            message,
+        })
+    }
+
     const handleCreateAccount = async () => {
         if (!account || !provider || !chainId) return false
-        const txData = await geb.contracts.proxyRegistry.populateTransaction['build()']()
-        const signer = provider.getSigner(account)
+
         try {
+            await sendConnectWalletEvent(account)
+            const txData = await geb.contracts.proxyRegistry.populateTransaction['build()']()
+            const signer = provider.getSigner(account)
```

Comment:
Why is this moved inside the try-catch?

Reply:
because sendConnectWalletEvent is a network call and signature request and both can fail